polkadot/runtime/parachains/src/inclusion/mod.rs

```diff
@@ -882,13 +882,31 @@ impl<T: Config> Pallet<T> {
 			// Block number of candidate's inclusion.
 			let now = <frame_system::Pallet<T>>::block_number();
 
-			weight.saturating_add(<paras::Pallet<T>>::schedule_code_upgrade(
+			match <T as paras::Config>::PreCodeUpgrade::pre_code_upgrade(
 				receipt.descriptor.para_id,
-				new_code,
-				now,
-				&config,
-				SetGoAhead::Yes,
-			));
+				new_code.clone(),
+				UpgradeRequirements::EnforceRequirements,
+			) {
+				Ok(consumed_weight) => {
+					weight.saturating_accrue(consumed_weight);
+
+					weight.saturating_accrue(<paras::Pallet<T>>::schedule_code_upgrade(
+						receipt.descriptor.para_id,
```

Comment:
On a different note: Would be good to check (if you haven't done so), whether just not doing the upgrade is sound. We can definitely not do it, as this also happens if pre-checking fails, but do we need to report anything back to the parachain? I would guess not, but I am not 100% sure.

Reply:
> whether just not doing the upgrade is sound.

I am a bit confused, why wouldn't we do the upgrade if it passed`pre_code_upgrade` successfully?

Reply:
Because it failed the pre-checking phase by validators. (They had problems compiling the artifact)

Reply:
I am still uncertain about what you are referring to here. The line you pointed to above schedules a code upgrade. From here we cannot determine whether the outcome of the scheduled upgrade will be successful or not(i.e. whether it passes pre-checking or not).